### DISCUSSION TITLE
Udemy videos not downloading.

### DISCUSSION THREAD

**Discussion by swapnilsinhrajput** [aclap-dev/video-downloadhelper#49] [Help and Q&A]
### Your question

I can attach a video of the problem. The extension downloaded one video then it had qr code on it I bought license to remove it and it worked. The first video was downloaded. But now every next video is only of size 500kb 1mb max and the video is not getting downloaded it does not work. Please help me with this problem. I bought license in hope to download more videos but here the case is not even single video is downloadable after the first video.

**Comment by swapnilsinhrajput** [comment]
So what I figured out is the app is able to download only one video from udemy from each section. I.e only the first video of each section after that other videos are not able to get downloaded please help.

**Comment by paulrouget** [comment]
> Addon & CoApp version and details
> No response

I need these information to assist you.

Also check if the beta version helps: https://www.downloadhelper.net/beta

**Reply by swapnilsinhrajput** [reply]
where can I find that information?

**Reply by GrampaWildWilly** [reply]
https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

**Reply by swapnilsinhrajput** [reply]
version: 8.1.6.0
beta: false
build date: 2024-01-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: accepted
key (only first 16 fisrt characters): 3b5abe0d171c4bf1
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.15",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\wHitedEviLL",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by paulrouget** [comment]
This probably works better in version v8.2 (which is in beta at the moment). You can wait a few days before the release, or try it yourself: https://downloadhelper.net/beta

**Reply by swapnilsinhrajput** [reply]
with beta I am getting this error when video is downloaded to 99%.


Native port disconnected
Error: Native port disconnected
    at bs.cleanup (chrome-extension://pfoiagbblcbmognbkekfpodpidedkmcc/background/main.js:1:43502)
    at chrome-extension://pfoiagbblcbmognbkekfpodpidedkmcc/background/main.js:1:42659

**Reply by GrampaWildWilly** [reply]
I'm going to make a guess here.  Other users have reported mysterious errors like this & it turned out to be their antivirus & firewall conspiring to interfere with the proper operation of the CoApp.  You should check to see if your antivirus has quarantined any part of the CoApp or otherwise prevented it from executing.  You should also check whether your firewall has blocked any part of the CoApp.  These would be false positives that the antivirus should not be throwing up.  We all know there is no virus in VDH/CoApp & that it's a perfectly benign application that should be allowed to run.  So find your CoApp directory.  You chose one or you allowed the CoApp installer to take the default.  If you don't know where that directory is, run the CoApp installer again until it gets to the point where it shows you what directory it will install in.  Make note of that directory, then cancel the installer.  Now, go into your antivirus & find where you can tell it to make an exception for a directory.  Tell it to make an exception for the CoApp directory you just found.  Then do the same thing in your firewall.  Let's see if that gets you going.  Some people have reported that their antivirus & their firewall just silently blocked the CoApp.  So if you're not actually getting errors popping up, that is bad design on the part of the antivirus/firewall, but there's nothing you can do about it other than complain to the vendor in question.  VDH/CoApp is just the victim here.

For more on the subject, read this: https://github.com/aclap-dev/video-downloadhelper/discussions/46

**Reply by swapnilsinhrajput** [reply]
The videos are getting downloaded now but the video is all glitched & blurred. The size of the video is good but when we open video inside no audio and no video just glitches and bizarre screen. Any solution for this? The first video from udemy gets downloaded perfectly it is the problem from 2nd video and afterwards.

**Reply by GrampaWildWilly** [reply]
If you post a screenshot from VLC while trying to play one of these videos, I could be more certain.  But guessing based on your description, the second & subsequent parts of the course are protected by DRM.  You can verify this for yourself by disabling DRM in the browser settings -- browser settings, not VDH settings -- & then reload the web page for the lesson that is failing.  Then try to play the lesson in the web page.  If it fails to play & the browser pops up a message about content on the page needing DRM, then you're out of luck.  VDH has never, cannot now, & will never download content protected by DRM.  We just had this confirmed yet again for the umpteenth time over here: https://github.com/aclap-dev/video-downloadhelper/discussions/50.  Your only recourse at this point for getting content such as this would seem to be a screen recorder.

IF

I'm guessing.  That's only if the content is protected by DRM.  Let's determine that first & go from there.

**Reply by swapnilsinhrajput** [reply]
Yes I think it is DRM. any suggestion for screen recorder? 
![Screenshot (211)](https://github.com/aclap-dev/video-downloadhelper/assets/132950043/4706056c-86c9-42c4-839c-8f3b939c69ed)

**Reply by GrampaWildWilly** [reply]
Yeah, that's classic DRM for you.  So forget about downloading it with any tool, not just VDH.  It's weird that they aren't consistent.  You'd expect to find either no part of any course is protected by DRM, or every part of every course is protected by DRM, not this some have it & some don't.  It seems they are doing this on purpose to annoy people who want to download the stuff.

As for screen recorders, I would just do a web search & see what looks good to you.  I have experience with only 1 such app, so anything I say about screen recorders is most definitely not informed.  I've used OBS but I'm not saying it's the best there is.  You need to do your own research.
